ci/infra/testrunner/tests/conftest.py

```diff
@@ -81,3 +81,60 @@ def kubectl(conf, target):
 def platform(conf, target):
     platform = platforms.get_platform(conf, target)
     return platform
+
+
+@pytest.fixture
+def cluster_ready(kubectl):
+    def func():
+        wait(kubectl.run_kubectl, 'wait --for=condition=Ready node --all --timeout=1m', wait_delay=60, wait_timeout=300, wait_backoff=30, wait_retries=5)
+        wait(kubectl.run_kubectl, 'wait --timeout=1m --for=condition=Ready pods --all --namespace=kube-system --field-selector=status.phase!=Succeeded', wait_delay=60, wait_timeout=300, wait_backoff=30, wait_retries=5, wait_allow=(RuntimeError))
+
+    return func
+
+
+@pytest.fixture
+def provision_master(request, conf, platform):
+    def cleanup():
+        if platform.get_num_nodes('master') > conf.master.count:
+            platform.remove_nodes('master', 1)
+
+    request.addfinalizer(cleanup)
+
+    platform.add_nodes('master', 1)
+
+
+@pytest.fixture
+def provision_worker(request, conf, platform):
+    def cleanup():
+        if platform.get_num_nodes('worker') > conf.worker.count:
+            platform.remove_nodes('worker', 1)
+
+    request.addfinalizer(cleanup)
+
+    platform.add_nodes('worker', 1)
+
+
+@pytest.fixture
+def joined_master(request, conf, skuba, cluster_ready, provision_master):
+    def cleanup():
+        if skuba.num_of_nodes('master') > conf.master.count:
+            skuba.node_remove(role='master', nr=skuba.num_of_nodes('master') - 1)
+
+    request.addfinalizer(cleanup)
+
+    skuba.node_join(role='master', nr=skuba.num_of_nodes('master'))
+
+    cluster_ready()
+
+
+@pytest.fixture
```

Comment:
Unless we are reusing this logic in multiple tests, I think it shouldn't be defined as a fixture. IMHO this logic is part of the test itself.

Reply:
removed